src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1472,7 +1472,7 @@ private BoundExpression BindAssignment(AssignmentExpressionSyntax node, BindingD
 
         private static BindValueKind GetRequiredRHSValueKindForRefAssignment(BoundExpression boundLeft)
         {
-            var rhsKind = BindValueKind.RefersToLocation;
+            var rhsKind = BindValueKind.RefersToLocation | BindValueKind.RValue;
```

Comment:
> | BindValueKind.RValue [](http://example.com/codeflow?start=56&length=23)

The rational behind this change is not obvious, Could you elaborate, please?

Reply:
I'm not sure I understand the original code correctly, but it seems to me that RHS of assignment should be an RValue. (And it improved the confusing error messages.)

Reply:
> but it seems to me that RHS of assignment should be an RValue.

RValue cannot be used on the right hand side of a a ref assignment. This method is specific for "RefAssignment". 

> And it improved the confusing error messages.

But it is likely to have an effect far beyond the diagnostics.